embedded_rtic/examples/sonars.rs

```diff
@@ -0,0 +1,448 @@
+                        cx.local.times[2] = time;
+                    } else {
+                        let distance3 = time
+                            .checked_duration_since(cx.local.times[2])
+                            .unwrap()
+                            .to_micros()
+                            / 56;
+                        //trace!("distance 3 before sending: {:?}", distance3);
+                        cx.local.times[2] = zero;
+                        match cx.local.senders[2].try_send(distance3 as u32) {
+                            Ok(_) => {}
+                            _ => {
+                                error!("Distance RIGHT did not fit in to the buffer, scheduling is probably broken.");
+                                // This is not fatal so we simply continue with
+                                // our life
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        // Clear interrupt pending bits.
+        cx.shared.gpiote.lock(|gpiote| {
+            gpiote.reset_events();
+            gpiote.channel0().clear();
+            gpiote.channel1().clear();
+            gpiote.channel2().clear();
+        });
+    }
+
+    #[task(priority = 2, local = [trig], shared = [gpiote])]
+    /// Send a small pulse to the sonars.
+    ///
+    /// The sonars will then notify us in echo when the sound waves are
+    /// recieved.
+    async fn trigger1(cx: trigger1::Context) {
+        // Set high is always valid.
+        cx.local.trig.set_high().unwrap();
+
+        let now = Mono::now();
+        Mono::delay_until(now + 20.micros()).await;
+
+        // Set low is always valid.
+        cx.local.trig.set_low().unwrap();
+    }
+
+    #[task(priority = 2, local = [trig2], shared = [gpiote])]
+    /// Send a small pulse to the sonars.
+    ///
+    /// The sonars will then notify us in echo when the sound waves are
+    /// recieved.
+    async fn trigger2(cx: trigger2::Context) {
+        // Set high is always valid.
+        cx.local.trig2.set_high().unwrap();
+
+        let now = Mono::now();
+        Mono::delay_until(now + 20.micros()).await;
+
+        // Set low is always valid.
+        cx.local.trig2.set_low().unwrap();
+    }
+
+    #[task(priority = 2, local = [trig3], shared = [gpiote])]
+    /// Send a small pulse to sonar 3.
+    ///
+    /// The sonar will then notify us in echo when the sound wave is recieved.
+    async fn trigger3(cx: trigger3::Context) {
+        // Set high is always valid.
+        cx.local.trig3.set_high().unwrap();
+
+        let now = Mono::now();
+        Mono::delay_until(now + 20.micros()).await;
+
+        // Set low is always valid.
+        cx.local.trig3.set_low().unwrap();
+    }
+
+    #[task(priority = 1,local = [receiver_forward])]
+    /// Re-spawn trigger after every new distance is correctly measured.
+    async fn trigger_trampoline1(cx: trigger_trampoline1::Context) {
```

Comment:
We do not need 3 different software tasks for these, it is fine for now but we can just pass the receivers as function arguments and spawn the same function with 3 different arguments. But for this example, it is fine.